nltk/metrics/distance.py

```diff
@@ -182,18 +186,62 @@ def custom_distance(file):
             labelA, labelB, dist = l.strip().split("\t")
             labelA = frozenset([labelA])
             labelB = frozenset([labelB])
-            data[frozenset([labelA,labelB])] = float(dist)
-    return lambda x,y:data[frozenset([x,y])]
+            data[frozenset([labelA, labelB])] = float(dist)
+    return lambda x, y: data[frozenset([x, y])]
+
+
+def jaro_winkler_distance(s1, s2, p=0.1):
+    """ Details about this implementation can be found at
+    https://en.wikipedia.org/wiki/Jaro-Winkler_distance
+dist_bound = type int, upper bound of distance for being a matched character
+mtch = type int, no.of matched characters in s1 and s2
+t_cnt= half of the number of transpositions between s1 and s2
+j_sim = type float ,jaro similarity between s1 and s2
+p = type float , weight parameter with a standard value of 0.1
+l_cnt = type int, common prefix at the start of the string (max val = 4)
+jw_sim= type float, jaro winkler similarity between s1 and s2
+jw_dist = type float, it is equal to 1 - jw_sim """
+    s1 = s1.lower()
+    s2 = s2.lower()
```

Comment:
As commented above, if the input is a sequence, this would throw an AttributeError.

Reply:
If `.lower()` is the intention, then adding it as an kwarg in the function arguments would be better, i.e. 

```
def jaro_winkler_distance(s1, s2, p=0.1, lowercase=True):
...

    if lowercase:
        s1, s2 = s1.lower(), s2.lower()

...
```